client/ayon_core/hosts/nuke/api/utils.py

```diff
@@ -117,19 +118,22 @@ def get_colorspace_list(colorspace_knob):
     Returns:
         list: list of strings names of profiles
     """
-
-    all_clrs = list(colorspace_knob.values())
-    reduced_clrs = []
-
-    if not colorspace_knob.getFlag(nuke.STRIP_CASCADE_PREFIX):
-        return all_clrs
-
-    # strip colorspace with nested path
-    for clrs in all_clrs:
-        clrs = clrs.split('/')[-1]
-        reduced_clrs.append(clrs)
-
-    return reduced_clrs
+    results = []
+
+    # This pattern is to match with roles which uses an indentation and
+    # parentheses with original colorspace. The value returned from the
+    # colorspace is the string before the indentation, so we'll need to
+    # convert the values to match with value returned from the knob,
+    # ei. knob.value().
+    pattern = r".*\t.* \(.*\)"
+    for colorspace in nuke.getColorspaceList(colorspace_knob):
+        match = re.search(pattern, colorspace)
+        if match:
+            results.append(colorspace.split("\t")[0])
```

Comment:
```suggestion
            results.append(colorspace.split("\t", 1)[0])
```
This should be faster since it can stop scanning the string after the first tab found.

I also wonder whether the regex matching is really needed to begin with - not a huge issue but does leave me wondering. Since you're matching anyway you could also do:

```python
import re


names = [
    # Non tabbed
    "mycolorspace",
    "ACES - ACEScg",
    # Tabbed
    "compositing_log	compositing_log (Input - ADX - ADX10)",
    "data	data (Utility - Raw)",
    "default	default (ACES - ACES2065-1)",
    "matte_paint	matte_paint (Utility - sRGB - Texture)",
    "reference	reference (Utility - Raw)",
    "rendering	rendering (ACES - ACEScg)",
    "scene_linear	scene_linear (ACES - ACEScg)",
    "texture_paint	texture_paint (ACES - ACEScc)"
]


for colorspace in names:
    match = re.match(r"(.*)\t.* \(.*\)", colorspace)
    if match:
        colorspace = match.group(1)
    print(colorspace)
```

Also, note that regex is greedy in this case so MIGHT match multiple tabs with the first `.*`. Not a big problem, just wanted to note that.

Reply:
> Since you're matching anyway you could also do:

Any benefit over the existing code?

Reply:
> Any benefit over the existing code?

Speed, the match speed of the regex will be roughly equivalent - and thus you're not doing the extra string parsing for `.split()`

Plus it's clearer that you're matching the regex to ensure you get the right part - now the mixture of matching and then splitting might mean that you're not matching and splitting on the same point of the string.

In this case a string with more than one tab will be matched to the last tab found in the string, whereas the `split` call wouldn't care.

Any not too big of an issue - I guess it was me just trying to micro-optimize and be explicit about the behavior.